contrib_bots/lib/github_detail.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import print_function
+
+import re
+import os
+import sys
+import logging
+import six.moves.configparser
+
+import requests
+
+class GithubHandler(object):
+    '''
+    This bot provides details on github issues and pull requests when they're
+    refrenced in the chat.
+    '''
+
+    config_path = os.path.expanduser('~/.contrib_bots/github_detail.ini')
+    GITHUB_ISSUE_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{repo}/issues/{id}'
+    GITHUB_PULL_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{reepo}/pulls/{id}'
+
+    def __init__(self):
+        config = six.moves.configparser.ConfigParser()
+        config.readfp(open(self.config_path))
+        if config.get('Github', 'owner'):
+            self.owner = config.get('Github', 'owner')
+        else:
+            # Allowing undefined default repos would require multiple triage_message regexs.
+            # It's simpler to require them to be defined.
+            sys.exit('Default owner not defined')
+
+        if config.get('Github', 'repo'):
+            self.repo = config.get('Github', 'repo')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To refrence an issue or pr type @github-detail then anytime in the"
+                " message it's id, for example:\n"
+                "@**Github detail** #3212 zulip/#3212 zulip/zulip#3212\n"
+                "The default owner is {} and the default repo is {}.".format(self.owner, self.repo))
+
+    def triage_message(self, message, client):
+        # type: () -> bool
+        # Check the message contains a username mention, an issue idi
+        # or 'help', and that we're not replying to another bot.
+        regex = "(?:@(?:\*\*){}).+?(?:#\d+)|(?:help)".format(re.escape(client.full_name))
```

Comment:
That's because the regex is structured to accept username mentions which in zulip are formatted like @**username**. The merits of this have already been discussed.